internal/provisioner/provisioner.go

```diff
@@ -0,0 +1,70 @@
+package provisioner
+
+import (
+	"github.com/mattermost/mattermost-cloud/internal/api"
+	"github.com/mattermost/mattermost-cloud/internal/store"
+	"github.com/mattermost/mattermost-cloud/internal/supervisor"
+	"github.com/mattermost/mattermost-cloud/internal/tools/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+type ClusterProvisionerOption struct {
+	kopsProvisioner *KopsProvisioner
+	eksProvisioner  *EKSProvisioner
+}
+
+func (c ClusterProvisionerOption) GetClusterProvisioner(provisioner string) supervisor.ClusterProvisioner {
+	if provisioner == "eks" {
+		return c.eksProvisioner
+	}
+
+	return c.kopsProvisioner
+}
+
+type Provisioner struct {
+	ClusterProvisionerOption
+	params         ProvisioningParams
+	resourceUtil   *utils.ResourceUtil
+	backupOperator *BackupOperator
+	store          *store.SQLStore
+	logger         log.FieldLogger
+	kubeOption     ClusterProvisionerOption
+}
+
+var _ supervisor.ClusterInstallationProvisioner = (*Provisioner)(nil)
```

Comment:
What am I looking at here?

Reply:
This is just a nice way to assert that `struct` implements expected set of interfaces.

Reply:
Huh, interesting. Never seen this before.